mod_ood_proxy/lib/ood/nginx_stage.lua

```diff
@@ -74,12 +81,59 @@ end
 --[[
   capture2e
 
-  Give a string for stdin, get a string for merged stdout and stderr
+  Give a string for a command, get a string for merged stdout and stderr
 --]]
 function capture2e(cmd)
   return capture2(cmd .. " 2>&1")
 end
 
+--[[
+  capture2e_with_stdin
+
+  Give a command and some standard in, get a string for merged stdout and stderr
+--]]
+function capture2e_with_stdin(cmd, stdin)
+  local output_file = os.tmpname()
```

Comment:
Since we are back to using temporary filenames, and every single PUN launch will now use this, should we do io.tmpfile() instead? Also, this file we should ensure the permissions to be 600 before writing sensitive data to this.

Reply:
When I was testing this, I saw it was creating 600 files, though that could be platform/configuration dependent.  You think a `os.execute(chmod ... )` is enough?

Looks like I was using `os.tmpname` because it's actually a string, so don't have to cast it.
```lua
> t = os.tmpname()
> os.execute('chmod 600 ' .. t .. ' >/dev/null 2>&1')
true	exit	0
> f = io.tmpfile()
> os.execute('chmod 600 ' .. f .. ' >/dev/null 2>&1')
stdin:1: attempt to concatenate a FILE* value (global 'f')
stack traceback:
	stdin:1: in main chunk
	[C]: in ?
```
Then I'm not sure how to cast it to a string.
```lua
> tostring(f)
file (0x56056fb7ea90)
> tostring(t)
/tmp/lua_8GVcnl
```

Reply:
Maybe we can rely on `io.tmpfile()` to create 600 files. It is either that or introduce a tiny race condition with the use of `os.tmpname`.